src/App.tsx

```diff
@@ -1,17 +1,132 @@
-import React from 'react';
-import './App.css';
-
+import React, { useState } from 'react';
+import todos from './api/todos';
 import users from './api/users';
+import './App.scss';
+import { TodoList } from './components/TodoList/TodoList';
 
 const App: React.FC = () => {
+  const preparedTodos = todos.map(todo => ({
+    ...todo,
+    user: users.find(user => todo.userId === user.id) || null,
+  }));
+
+  const [todoTitle, setTodoTitle] = useState('');
+  const [userSelect, setUseSelect] = useState('');
+  const [todoList, setTodoList] = useState([...preparedTodos]);
+  const [subbmited, setSubbmited] = useState(false);
+
+  const addTodo = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    const userId = users.find(user => userSelect === user.username)?.id || 0;
+    const todoId = todoList.length + 1;
+
+    if (!userSelect || !todoTitle.trim()) {
+      setSubbmited(true);
+    } else {
+      setTodoList((prev) => [
+        ...prev,
+        {
+          user: users.find(user => userSelect === user.username) || null,
+          userId,
+          id: todoId,
+          title: todoTitle,
+          completed: false,
+        },
+      ]);
+      setSubbmited(false);
+      setTodoTitle('');
+      setUseSelect('');
+    }
+  };
+
+  const onInputChange = (ev: React.ChangeEvent<HTMLInputElement>) => {
+    const value = ev.target.value
+      .replace(/([^a-zA-Z0-9а-яА-Я ])/g, '');
+
+    setTodoTitle(value);
+  };
+
   return (
     <div className="App">
-      <h1>Add todo form</h1>
+      <h1>Add new ToDo</h1>
 
       <p>
         <span>Users: </span>
         {users.length}
       </p>
+
+      <form
+        className="form-add-todo"
+        onSubmit={addTodo}
+      >
+        <label
+          className="form-add-todo__label-input"
+        >
+          Title
+          {' '}
+          <input
+            type="text"
+            className="form-add-todo__label-input__textarea"
+            placeholder="Write a title for your ToDo"
+            value={todoTitle}
+            onChange={onInputChange}
+          />
+        </label>
+        {!todoTitle.trim() && subbmited && (
+          <span
+            style={{
+              marginLeft: '5px',
+              color: 'red',
+              fontWeight: '700',
+            }}
```

Comment:
don't add inline styles, add classes instead

Reply:
use this if you need conditionally set styles 
https://www.npmjs.com/package/classnames